plugins/Live/Model.php

```diff
@@ -43,20 +43,123 @@ public function queryLogVisits($idSite, $period, $date, $segment, $offset, $limi
             $limit++;
         }
 
-        list($sql, $bind) = $this->makeLogVisitsQueryString($idSite, $period, $date, $segment, $offset, $limit, $visitorId, $minTimestamp, $filterSortOrder);
+        // If no other filter, only look at the last 24 hours of stats
+        if (empty($visitorId)
+            && empty($limit)
+            && empty($offset)
+            && empty($period)
+            && empty($date)
+        ) {
+            $period = 'day';
+            $date = 'yesterdaySameTime';
+        }
+
+        list($dateStart, $dateEnd) = $this->getStartAndEndDate($idSite, $period, $date);
+
+        $virtualDateEnd = $dateEnd;
+        if (empty($dateEnd)) {
+            $virtualDateEnd = Date::now()->addDay(1); // matomo always adds one day for some reason
+        }
+
+        $virtualDateStart = $dateStart;
+        if (empty($virtualDateStart)) {
+            $virtualDateStart = Date::factory(Date::FIRST_WEBSITE_TIMESTAMP); // matomo always adds one day for some reason
+        }
+
+        $queries = [];
+        $hasStartEndDateMoreThanOneDayInBetween = $virtualDateStart && $virtualDateStart->addDay(1)->isEarlier($virtualDateEnd);
+        if ($limit
+            && $hasStartEndDateMoreThanOneDayInBetween
+        ) {
+            if ($hasStartEndDateMoreThanOneDayInBetween) {
+                $virtualDateEnd = $virtualDateEnd->subDay(1);
+                $queries[] = array($virtualDateEnd, $dateEnd); // need to use ",endDate" in case endDate is not set
+            }
```

Comment:
is this `if` needed? Above you check for `$limit && $hasStartEndDateMoreThanOneDayInBetween`, so `$hasStartEndDateMoreThanOneDayInBetween` should be true.

Reply:
Good point, will remove it